pkg/operator/staticpod/controller/installer/installer_controller_test.go

```diff
@@ -884,6 +885,37 @@ func TestCreateInstallerPodMultiNode(t *testing.T) {
 			updateStatusErrors:   []error{errors.NewInternalError(fmt.Errorf("unknown"))},
 			expectedSyncError:    []bool{true},
 		},
+		{
+			name:                    "three nodes, 2 not updated, one with failure in last revision which is no longer available",
+			latestAvailableRevision: 3,
+			nodeStatuses: []operatorv1.NodeStatus{
+				{
+					NodeName:        "test-node-0",
+					CurrentRevision: 1,
+				},
+				{
+					NodeName:           "test-node-1",
+					CurrentRevision:    1,
+					LastFailedRevision: 2,
+				},
+				{
+					NodeName:        "test-node-2",
+					CurrentRevision: 1,
+				},
+			},
+			staticPods: []*corev1.Pod{
+				newStaticPod(mirrorPodNameForNode("test-pod", "test-node-1"), 1, corev1.PodRunning, true),
+				newStaticPod(mirrorPodNameForNode("test-pod", "test-node-2"), 2, corev1.PodRunning, true),
```

Comment:
that one had failed according to `LastFailedRevision`. Shouldn't it still be at 1?

Reply:
The tests never actually check this revision number. I've changed the test, as I've realized that none of these should exist in the scenario (the namespace was deleted).